pkg/snmp/traps/config.go

```diff
@@ -60,12 +82,90 @@ func (c *Config) Addr() string {
 	return fmt.Sprintf("%s:%d", c.BindHost, c.Port)
 }
 
-// BuildV2Params returns a valid GoSNMP SNMPv2 params structure from configuration.
-func (c *Config) BuildV2Params() *gosnmp.GoSNMP {
-	return &gosnmp.GoSNMP{
-		Port:      c.Port,
-		Transport: "udp",
-		Version:   gosnmp.Version2c,
-		Logger:    gosnmp.NewLogger(&trapLogger{}),
+func (c *Config) getAgentHostname() string {
+	hostname, err := util.GetHostname(context.TODO())
+	if err != nil {
+		// this scenario is not likely to happen since
+		// the agent cannot start without a hostname
+		hostname = "unknown-datadog-agent"
 	}
+	return hostname
+}
+
+// BuildAuthoritativeEngineID creates a engineID byte-array from the OS hostname
+func (c *Config) BuildAuthoritativeEngineID() [28]byte {
+	engineID := [28]byte{}
+	// First byte is always 0x80
+	// Next four bytes are the Private Enterprise Number (set to an invalid value here)
```

Comment:
Do we set it to the valid number elsewhere?

Reply:
It's valid enough so that it works fine. Just that in theory, the four bytes are expected to represent the PEN identifier for Datadog which we don't have: https://www.iana.org/assignments/enterprise-numbers/enterprise-numbers
`0xFFFFFFFF` is a value that doesn't represent any other company.